src/store/index.js

```diff
@@ -502,50 +502,40 @@ export function sendCssToFrame(value, currentField) {
     })
   })
 
-  // Checks if there is a field that has a dependency on the value that has been changed
-  configurations.forEach((config) => {
-    config.variables.forEach((variable) => {
-      variable.fields.forEach((field) => {
-        const inheritingVariable = isInheriting(field.default)
+  const dependenciesFound = findDependencies(configurations, currentField)
 
-        const fieldStyles = field.styles || []
-        fieldStyles.forEach((style) => {
-          // Check if the names match
-          if (!inheritingVariable || inheritingVariable !== currentField.name) {
-            return
-          }
-
-          savedWidgetFound = _.find(state.themeInstance.settings.widgetInstances, (widget) => {
-            return !!widget.values[field.name]
-          })
-          localSavedWidgetFound = _.find(state.savedFields.widgetInstances, (widget) => {
-            return !!widget.values[field.name]
-          })
+  dependenciesFound.forEach((field) => {
+    const fieldStyles = field.styles || []
+    fieldStyles.forEach((style) => {
+      savedWidgetFound = _.find(state.themeInstance.settings.widgetInstances, (widget) => {
+        return !!widget.values[field.name]
+      })
+      localSavedWidgetFound = _.find(state.savedFields.widgetInstances, (widget) => {
+        return !!widget.values[field.name]
+      })
 
-          const savedValues = getSavedValue(field, true)
+      const savedValues = getSavedValue(field, true)
 
-          if (savedValues.generalSavedValue
-            || savedValues.generalLocalSavedValue
-            || savedValues.widgetSavedValue
-            || savedValues.widgetLocalSavedValue) {
-            return
-          }
+      if (savedValues.generalSavedValue
```

Comment:
- what is a `general` saved value?
what is a `local` saved value?

Reply:
@squallstar In the context of everything `general` is referred as the settings for all components. So when it says `widget` that are settings specifically for a widget.

In regards to `local` means settings that are "saved" locally in memory. When it doesn't say `local` are the settings that are already stored on the DB as part of the theme instance settings.